### PR TITLE
webdriver: add actual capabilities processing of experitest response (#4501)

### PR DESCRIPTION
**Pull request by andy-schulz**
## Proposed changes

experitest returns the actuial capabilities in a format not processed by `isMobile` and `isW3C`. The following change contains:

add "device.name" to isW3C()
add "device.name", "device.orientation", "device.type" to isMobile()
add experitest example response
add new isMobile test spec
add experitest check to isW3C test spec

See #4501 for details

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/webdriver/src/utils.js [hunk 1]
```diff
@@ -218,13 +218,18 @@ export function isChrome (caps) {
...
 export function isMobile (caps) {
     return Boolean(
-        (typeof caps['appium-version'] !== 'undefined') ||
+        (typeof caps['appium-version'] !== 'undefined') || (typeof caps['appiumVersion'] !== 'undefined') ||
         (typeof caps['device-type'] !== 'undefined') || (typeof caps['deviceType'] !== 'undefined') ||
         (typeof caps['device-orientation'] !== 'undefined') || (typeof caps['deviceOrientation'] !== 'undefined') ||
         (typeof caps.deviceName !== 'undefined') ||
+        // experitest returns actual device capabilities in the form "device.name" | "device.model" etc.
+        (typeof caps['device'] !== 'undefined' && (
+            (typeof caps['device.name'] !== 'undefined') ||
```
Comment (mgrybyk) [inline packages/webdriver/src/utils.js]: I see lots of common when checking properties like deviceName, device.name, device-name.
What do you think on adding function that accepts caps, device, name and verifies all 3 combinations.

File: packages/webdriver/src/utils.js [hunk 2]
```diff
@@ -218,13 +218,18 @@ export function isChrome (caps) {
  */
 export function isMobile (caps) {
     return Boolean(
-        (typeof caps['appium-version'] !== 'undefined') ||
+        (typeof caps['appium-version'] !== 'undefined') || (typeof caps['appiumVersion'] !== 'undefined') ||
         (typeof caps['device-type'] !== 'undefined') || (typeof caps['deviceType'] !== 'undefined') ||
         (typeof caps['device-orientation'] !== 'undefined') || (typeof caps['deviceOrientation'] !== 'undefined') ||
         (typeof caps.deviceName !== 'undefined') ||
+        // experitest returns actual device capabilities in the form "device.name" | "device.model" etc.
```
Comment (christian-bromann) [inline packages/webdriver/src/utils.js]: I would not like to include experitest specific capabilties, only caps that are defined by [Appium](http://appium.io/docs/en/writing-running-appium/caps/)

File: packages/webdriver/src/utils.js [hunk 3]
```diff
@@ -185,7 +185,7 @@ export function isW3C (capabilities) {
      *   (https://w3c.github.io/webdriver/#dfn-new-sessions)
      * - it is an Appium session (since Appium is full W3C compliant)
      */
-    const isAppium = capabilities.automationName || capabilities.deviceName
+    const isAppium = capabilities.automationName || capabilities.deviceName || (capabilities['device.name'])
```
Comment (christian-bromann) [inline packages/webdriver/src/utils.js]: `appium.name` is experitest specific, feel free to add `appiumVersion`

File: packages/webdriver/src/utils.js (hunk 2 above)
Comment (andy-schulz) [reply packages/webdriver/src/utils.js]: Do you mean I should remove all device. XYZ capabilities and just check for appiumVersion?

File: packages/webdriver/src/utils.js (hunk 3 above)
Comment (andy-schulz) [reply packages/webdriver/src/utils.js]: I did not get it. Do you mean device.name ? Shall I remove device.name and just test for appiumVersion. If this is the case ... Sure, will update the check.

File: packages/webdriver/src/utils.js [hunk 4]
```diff
@@ -185,7 +185,7 @@ export function isW3C (capabilities) {
      *   (https://w3c.github.io/webdriver/#dfn-new-sessions)
      * - it is an Appium session (since Appium is full W3C compliant)
      */
-    const isAppium = capabilities.automationName || capabilities.deviceName
+    const isAppium = capabilities.automationName || capabilities.deviceName || (capabilities['appiumVersion'])
```
Comment (christian-bromann) [inline packages/webdriver/src/utils.js]:
```suggestion
    const isAppium = capabilities.automationName || capabilities.deviceName || (capabilities.appiumVersion)
```

File: packages/webdriver/src/utils.js [hunk 5]
```diff
@@ -218,13 +218,13 @@ export function isChrome (caps) {
  */
 export function isMobile (caps) {
     return Boolean(
-        (typeof caps['appium-version'] !== 'undefined') ||
+        (typeof caps['appium-version'] !== 'undefined') || (typeof caps['appiumVersion'] !== 'undefined') ||
```
Comment (christian-bromann) [inline packages/webdriver/src/utils.js]:
```suggestion
        (typeof caps['appium-version'] !== 'undefined') || (typeof caps.appiumVersion !== 'undefined') ||
```

**Pull request merged**